core/src/main/resources/jenkins/security/ApiTokenProperty/resources.js

```diff
@@ -0,0 +1,128 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+function revokeToken(anchorRevoke){
+    var repeatedChunk = anchorRevoke.up('.repeated-chunk');
+    var tokenList = repeatedChunk.up('.token-list');
+    var confirmMessage = anchorRevoke.getAttribute('data-confirm');
+    var targetUrl = anchorRevoke.getAttribute('data-target-url');
+    
+    var inputUuid = repeatedChunk.querySelector('input.token-uuid-input');
+    var tokenUuid = inputUuid.value;
+
+    if(confirm(confirmMessage)){
+        new Ajax.Request(targetUrl, {
+            method: "post",
+            parameters: {tokenUuid: tokenUuid},
+            onSuccess: function(rsp,_) {
+                if(repeatedChunk.querySelectorAll('.legacy-token').length > 0){
+                    // we are revoking the legacy token
+                    var messageIfLegacyRevoked = anchorRevoke.getAttribute('data-message-if-legacy-revoked');
+                    
+                    var legacyInput = document.getElementById('apiToken');
+                    legacyInput.value = messageIfLegacyRevoked;
+                }
+                repeatedChunk.remove();
+                adjustTokenEmptyListMessage(tokenList);
+                
+            }
+        });
+    }
+
+    return false;
+}
+
+function saveApiToken(button){
+    if(button.hasClassName('request-pending')){
+        // avoid multiple requests to be sent if user is clicking multiple times
+        return;
+    }
+    button.addClassName('request-pending');
+    var targetUrl = button.getAttribute('data-target-url');
+    var repeatedChunk = button.up('.repeated-chunk ');
+    var tokenList = repeatedChunk.up('.token-list');
+    var nameInput = repeatedChunk.querySelector('[name="tokenName"]');
+    var tokenName = nameInput.value;
+    
+    new Ajax.Request(targetUrl, {
+        method: "post",
+        parameters: {"newTokenName": tokenName},
+        onSuccess: function(rsp,_) {
+            var json = rsp.responseJSON;
+            var errorSpan = repeatedChunk.querySelector('.error');
+            if(json.status === 'error'){
+                errorSpan.style.display = 'block';
+                errorSpan.innerHTML = json.message;
+
+                button.removeClassName('request-pending');
```

Comment:
Shouldn't this be done if no error happens too?

Reply:
actually not because in the `else` there is `button.remove();`